src/Builder/Encoder/OperatorEncoder.php

```diff
@@ -95,7 +98,7 @@ private function encodeAsGroup(GroupStage $value): stdClass
         return $this->wrap($value, $result);
     }
 
-    private function encodeAsObject(OperatorInterface $value): stdClass
+    private function encodeAsObject(OperatorInterface $value, bool $flat = false): stdClass
```

Comment:
The `$flat` behavior is already hold by `$value`.

I would modify the `wrap` method to skip if `$operator::getOperator()` returns `null`.